Public/Src/App/Bxl/BuildXLApp.cs

```diff
@@ -896,6 +896,8 @@ private AppResult RunWithLoggingScope(Func<PerformanceMeasurement, AppResult> ru
                         translatedLogDirectory = pathTranslator != null ? pathTranslator.Translate(logDirectory) : logDirectory;
                     }
 
+                    TranslateGlobalUntrackedScope();

```

Comment:
>TranslateGlobalUntrackedScope [](start = 20, length = 29)

Would it make more sense to do this in: `PopulateAndValidateConfiguration` since that is were we do all our configuration mutations for the engine..

Reply:
The configuration in taht function is properly mutable already so you can leave the interface readonly

---
In reply to: [295508556](https://github.com/microsoft/BuildXL/pull/488#discussion_r295508556) [](ancestors = 295508556)